gap/attr.gi

```diff
@@ -1787,30 +1787,23 @@ function(D)
   return D;
 end);
 
-InstallMethod(DigraphAddAllLoops, "for a digraph by out-neighbours",
-[IsDigraphByOutNeighboursRep],
+InstallMethod(DigraphAddAllLoops, "for a digraph", [IsDigraph],
 function(D)
-  local ismulti, C, list, v;
+  local C, v;
   if HasIsReflexiveDigraph(D) and IsReflexiveDigraph(D) then
     return D;
   fi;
-  ismulti := IsMultiDigraph(D);
   C := DigraphMutableCopyIfImmutable(D);
-  list := C!.OutNeighbours;
-  Assert(1, IsMutable(list));
   for v in DigraphVertices(C) do
-    if not v in list[v] then
-      Add(list[v], v);
-      if not ismulti then
-        SetDigraphEdgeLabel(C, v, v, 1);
-      fi;
+    if not IsDigraphEdge(D, v, v) then
```

Comment:
By way of explanation, we use `IsDigraphEdge(D, v, v)` because `D` might know, for example, its adjacency matrix and this is hopefully then used by `IsDigraphEdge`, whereas `C` is mutable so doesn't have any attributes.